### PR TITLE
Support password authentication for Trino engine

### PR DESCRIPTION
**Pull request by hddong**
### _Why are the changes needed?_

Support SSL for trino engine.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -897,6 +897,55 @@ object KyuubiConf {
       .stringConf
       .createOptional
 
+  val ENGINE_TRINO_CONNECTION_PASSWORD: OptionalConfigEntry[String] =
+    buildConf("kyuubi.session.engine.trino.connection.password")
+      .doc("The password used for connecting to trino cluster")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:902]: fill password in URL but not `beeline -p` is not intuitive i think

Comment (hddong) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:902]: > fill password in URL but not `beeline -p` is not intuitive i think

Yes, but Kyuubi server authentication uses `-P`. And the passwords of server and engine are different at most time.

File: docs/deployment/settings.md [hunk 2]
```diff
@@ -456,6 +456,13 @@ kyuubi.session.engine.startup.error.max.size|8192|During engine bootstrapping, i
 kyuubi.session.engine.startup.maxLogLines|10|The maximum number of engine log lines when errors occur during engine startup phase. Note that this max lines is for client-side to help track engine startup issue.|int|1.4.0
 kyuubi.session.engine.startup.waitCompletion|true|Whether to wait for completion after engine starts. If false, the startup process will be destroyed after the engine is started. Note that only use it when the driver is not running locally, such as yarn-cluster mode; Otherwise, the engine will be killed.|boolean|1.5.0
 kyuubi.session.engine.trino.connection.catalog|&lt;undefined&gt;|The default catalog that trino engine will connect to|string|1.5.0
+kyuubi.session.engine.trino.connection.keystore.password|&lt;undefined&gt;|The keystore password used for connecting to trino cluster|string|1.7.0
```
Comment (yaooqinn) [inline docs/deployment/settings.md:459]: remove session?

Comment (yaooqinn) [reply docs/deployment/settings.md:459]: Mostly,
- `kyuubi.session.engine` means how a session prepares or interacts with an engine.
- `kyuubi.engine` means how an engine itself works

Comment (hddong) [reply docs/deployment/settings.md]: Get it and modified.

Comment (pan3793) [reply docs/deployment/settings.md]: We should update the existing configurations to make them consistent, and maybe we can remove `connection.` as well?

```
    buildConf("kyuubi.engine.trino.url")
      .version("1.6.0")
      .withAlternative("kyuubi.session.engine.trino.connection.url")
      .stringConf
      .createOptional
```

**Review by pan3793 — APPROVED** [review]
Let's change the version to 1.8.0 and merge it

**Comment by pan3793** [comment]
Merged to master

**Pull request closed without merging**